pandas/tests/arrays/test_datetimelike.py

```diff
@@ -651,3 +651,38 @@ def test_array_interface(self, period_index):
         result = np.asarray(arr, dtype="S20")
         expected = np.asarray(arr).astype("S20")
         tm.assert_numpy_array_equal(result, expected)
+
+
+@pytest.mark.parametrize(
+    "array",
+    [
+        pd.TimedeltaIndex(["1 Day", "3 Hours", "NaT"])._data,
+        pd.date_range("2000-01-01", periods=3, freq="D")._data,
+        pd.period_range("2000-01-01", periods=3, freq="D")._data,
+    ],
+    ids=lambda x: type(x).__name__,
+)
+def test_nat_assignment_array(array):
+    expected = type(array)._from_sequence([pd.NaT, array[1], array[2]])
+
+    casting_nats = {
+        TimedeltaArray: [pd.NaT, np.timedelta64("NaT", "ns")],
+        DatetimeArray: [pd.NaT, np.datetime64("NaT", "ns")],
+        PeriodArray: [pd.NaT],
+    }[type(array)]
+    non_casting_nats = {
```

Comment:
this would be better to parameterize over the non_casting_nats and the casting_nats in a separate test. too much going on here.

Reply:
changed